elide-core/src/main/java/com/yahoo/elide/parsers/state/CollectionTerminalState.java

```diff
@@ -71,8 +71,11 @@ public CollectionTerminalState(Class<?> entityClass, Optional<PersistentResource
         DocumentProcessor includedProcessor = new IncludedProcessor();
         includedProcessor.execute(jsonApiDocument, collection, queryParams);
 
-        DocumentProcessor sortProcessor = new SortProcessor();
-        sortProcessor.execute(jsonApiDocument, collection, queryParams);
+        // if we are using pagination, then we have also applied sorting rules directly into the HQL, etc
+        if (requestScope.getPagination().isEmpty()) {
```

Comment:
I am not sure if this is the best condition. Basically, the intention was to only skip SortProcessing on the returned data set in the case where we were not paginating the data, since for HQL, or Criterion we can do this on the MySQL level (mariaDB, etc) and therefore take advantage of the C level system for sorting and other indexing, however, when should we fallback to using the document sorting? Redis? non-SQL dbs? Is there a good way to ensure this condition is met for the correct reasons?